airflow/dag_processing/processor.py

```diff
@@ -796,8 +818,16 @@ def _execute_task_callbacks(self, dagbag: DagBag | None, request: TaskCallbackRe
         if task:
             ti.refresh_from_task(task)
 
-        ti.handle_failure(error=request.msg, test_mode=self.UNIT_TEST_MODE, session=session)
-        self.log.info("Executed failure callback for %s in state %s", ti, ti.state)
+        if callback_type is TaskInstanceState.SUCCESS:
+            context = ti.get_template_context(session=session)
+            if not ti.task:
+                return
```

Comment:
Is this for type-checking? `get_template_context` already makes sure there is a task, so this check never returns. If this is only to convince Mypy, we can use either

```python
context["task"].on_success_callback
```

or

```python
if TYPE_CHECKING:
    assert ti.task
callbacks = ti.task.on_success_callback
```

instead.